src/pages/Questionnaire.tsx

```diff
@@ -60,67 +60,167 @@ const SubmitButton = styled(NavigationButton).attrs({ as: "a" })`
 
 // types
 
-enum Type {
+export enum Type {
   RADIO = "radio",
   CHECKBOX = "checkbox",
   RANGE = "range",
-  DROPDOWN = "dropdown",
 }
 
 interface Option {
   optionId: number;
   optionLabel: string;
+  points: number;
 }
 
-interface QuesProps {
+export interface Question {
   id: number;
   title: string;
   type: Type;
   options: Option[];
 }
 
+type Responses = Record<number, number | number[]>;
+
+// Function to calculate the score for each option - used inside calculateTotalScore function
+const calculateOptionScore = (
+  points: number,
+  numberOfOptions: number,
+  maxPointsPerQuestion: number
+) => {
+  return (points / numberOfOptions) * maxPointsPerQuestion;
+};
+
+// Function to calculate the total score when the user submits the questionnaire
+const calculateTotalScore = (questions: Question[], responses: Responses) => {
+  const numberOfQuestions = questions.length;
+  // Calculation to determine the max points allotted per question.
+  const maxPointsPerQuestion = 100 / numberOfQuestions;
+  let totalScore = 0;
+
+  questions.forEach((question) => {
+    const response = responses[question.id];
+    let questionScore = 0;
+
+    if (response !== undefined) {
+      // Check if the response is an array (checkbox input) before calculating the score
+      if (Array.isArray(response)) {
+        response.forEach((ele) => {
+          const option = question.options.find((opt) => opt.optionId === ele);
+          if (option) {
+            questionScore += calculateOptionScore(
+              option.points,
+              question.options.length,
+              maxPointsPerQuestion
+            );
+          }
+        });
+        // Score calculation for range and radio inputs
+      } else {
+        const option = question.options.find(
+          (opt) => opt.optionId === response
+        );
+        if (option) {
+          questionScore += calculateOptionScore(
+            option.points,
+            question.options.length,
+            maxPointsPerQuestion
+          );
+        }
+      }
+    }
+
+    // Ensure the score for this question does not exceed the max points per question
+    totalScore += Math.min(questionScore, maxPointsPerQuestion);
+  });
+
+  // Ensure the total score does not exceed 100
+  return Math.min(totalScore, 100);
+};
+
 export function Questionnaire() {
   const { questionId } = useParams<{ questionId: string }>();
   const navigate = useNavigate();
-  const [currentQuestion, setCurrentQuestion] = useState<number>(
-    questionId ? parseInt(questionId) : -1
+  const storedResponses = localStorage.getItem("questionnaire");
+
+  // State to keep track of the current question number
```

Comment:
we have to maintain a balance for adding comments. Comments are nice to add where we think the logic is complicated or we have to do something extra that may not be clear at the first glance.
There's no right or wrong answer to when you should add comments, just something to keep in mind.